src/ILCompiler.Compiler/src/Compiler/NameMangler.cs

```diff
@@ -345,5 +348,40 @@ private string ComputeMangledFieldName(FieldDesc field)
 
             return mangledName;
         }
+
+        private ImmutableDictionary<string, string> _mangledStringLiterals = ImmutableDictionary<string, string>.Empty;
+
+        public string GetMangledStringName(string literal)
```

Comment:
You could put that logic in a separate name sanitizer though. (It's one of the things that is at the back of my head since the sanitization rules are different for different scenarios and you might want that to be pluggable.)

I don't insist on addressing any of this though.